app/models/institution.rb

```diff
@@ -190,6 +190,10 @@ def complaints
     end
   end
 
+  def facility_map
+    InstitutionTree.build(facility_code, version)
```

Comment:
Here you pass in the `facility_code` and `version` attributes of an institution, and then immediately retrieve the institution using those attributes in the `build` method.  How about just passing the institution instead?
```suggestion
    InstitutionTree.build(self)
```